src/basho_bench_worker.erl

```diff
@@ -334,3 +349,27 @@ rate_worker_run_loop(State, Lambda) ->
         ExitReason ->
             exit(ExitReason)
     end.
+
+%% Not passing state API - expect non-list spec or error during new attempt
+init_generators(false, Config, Id, Module) ->
```

Comment:
It would be good to enforce the expectation of not getting a list by adding a guard like `when is_tuple(Config) ->`.